src/Rules/Generators/YieldTypeRule.php

```diff
@@ -65,18 +66,18 @@ public function processNode(Node $node, Scope $scope): array
 
 		$messages = [];
 		if (!$this->ruleLevelHelper->accepts($returnType->getIterableKeyType(), $keyType, $scope->isDeclareStrictTypes())) {
-			$messages[] = sprintf(
+			$messages[] = RuleErrorBuilder::message(sprintf(
 				'Generator expects key type %s, %s given.',
 				$returnType->getIterableKeyType()->describe(VerbosityLevel::typeOnly()),
 				$keyType->describe(VerbosityLevel::typeOnly())
-			);
+			))->line($node->getLine())->build();
```

Comment:
You can leave RuleErrorBuilder in (since given enough time, I want to all rules return this instance), but remove the line() method call.